Add zoom persistence

j7126 opened the pull request:
<!-- Please change the Answers in the table below
     to reflect the contents of your pull request. -->

| Q                 | A
| ----------------- | ---
| New feature?      | yes
| Breaking changes? | no
| Deprecations?     | no
| New tests added?  | not needed

### Description

Related to #1906

Adds a new setting to change the default zoom level under general settings.

fxha commented:
@j7126 @Jocs Should we save the current window zoom or allow to set a default value that may adjusted at runtime without saving? Any options?

Jocs commented:
>Should we save the current window zoom or allow to set a default value that may adjusted at runtime without saving? Any options?

Could we just add one zoom setting? Can it be modified either through shortcut keys or menus, or through the preference panel?

fxha commented:
This was my first thought too, all zoom shortcuts should set the preference value. Are there any benefits from using separated zoom settings?

Jocs commented:
>This was my first thought too, all zoom shortcuts should set the preference value. Are there any benefits from using separated zoom settings?

Separate zoom settings will increase user comprehension?

j7126 commented:
> Could we just add one zoom setting? Can it be modified either through shortcut keys or menus, or through the preference panel?

Yeah now that I think about it that would probably be better.

j7126 commented:
Alright, I have updated it. 

I have changed the name from 'default zoom' to 'zoom', and now all zoom events will update the settings, and zoom state will persist.

fxha commented on src/renderer/pages/app.vue:
```diff
@@ -186,6 +190,9 @@ export default {
     }, false)
 
     this.$nextTick(() => {
+      this.$watch('zoom', function (zoom) {
```
This should be moved to the `watch` property above.

j7126 replied:
fixed

fxha reviewed (approved):
LGTM, thanks!
